exporters/ostream/include/opentelemetry/exporters/ostream/metrics_exporter.h

```diff
@@ -0,0 +1,159 @@
+#pragma once
+
+#include <iostream>
+#include "opentelemetry/sdk/metrics/aggregator/exact_aggregator.h"
+#include "opentelemetry/sdk/metrics/aggregator/gauge_aggregator.h"
+#include "opentelemetry/sdk/metrics/aggregator/histogram_aggregator.h"
+#include "opentelemetry/sdk/metrics/exporter.h"
+#include "opentelemetry/sdk/metrics/record.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdkmetrics = opentelemetry::sdk::metrics;
+
+namespace exporter
+{
+namespace metrics
+{
+
+/**
+ * The OStreamMetricsExporter exports record data through an ostream
+ */
+class OStreamMetricsExporter final : public sdkmetrics::MetricsExporter
+{
+public:
+  /**
+   * Create an OStreamMetricsExporter. This constructor takes in a reference to an ostream that the
+   * export() function will send span data into.
+   * The default ostream is set to stdout
+   */
+  explicit OStreamMetricsExporter(std::ostream &sout = std::cout) noexcept;
+
+  sdkmetrics::ExportResult Export(const std::vector<sdkmetrics::Record> &records) noexcept override;
+
+private:
+  std::ostream &sout_;
+
+  /**
+   * Send specific data from the given AggregatorVariant based on what AggregatorKind
+   * it is holding. Each Aggregator holds data differently, so each have their own
+   * custom printing.
+   */
+  template <typename T>
+  void PrintAggregatorVariant(sdkmetrics::AggregatorVariant value)
+  {
+    auto agg     = nostd::get<std::shared_ptr<sdkmetrics::Aggregator<T>>>(value);
+    auto aggKind = agg->get_aggregator_kind();
+
+    if (agg != nullptr)
```

Comment:
Could we simply the flow here:
1. short circuit if `agg == nullptr`.
2. use `switch case` for the conditions.

So the flow would look like:
```c
if (!agg) return;
switch (aggKind) {
case x:
  // do something
  break;
}
```